### PR TITLE
translate content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md into Chinese

### PR DESCRIPTION
**Pull request by sarazqy**
translate content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md into Chinese

fixes: https://github.com/kubernetes/website/issues/35866

### CODE REVIEW

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+convenient. We're happy to announce that we recently [released
+v0.4.0](https://github.com/kubernetes-sigs/security-profiles-operator/releases/tag/v0.4.0)
+of the operator, which contains a ton of new features, fixes and usability
+improvements.
+-->
+
+[安全分析文件操作器 (SPO)](https://sigs.k8s.io/security-profiles-operator) 
+是一种树外 Kubernetes 增强功能，用于管理 [seccomp](https://en.wikipedia.org/wiki/Seccomp)、
+[SELinux](https://en.wikipedia.org/wiki/Security-Enhanced_Linux) 和
+[AppArmor](https://en.wikipedia.org/wiki/AppArmor) 配置文件更简单、更方便。
```
Comment (Sea-n) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:37]: We can use Mandarin version Wikipedia articles.

```suggestion
[SELinux](https://zh.wikipedia.org/wiki/%E5%AE%89%E5%85%A8%E5%A2%9E%E5%BC%BA%E5%BC%8FLinux) 和
[AppArmor](https://zh.wikipedia.org/wiki/AppArmor) 配置文件更简单、更方便。 
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+repository](https://quay.io/organization/security-profiles-operator).
+The selinuxd repository has moved as well to [the containers GitHub
+organization](https://github.com/containers/selinuxd).
+-->
+### SELinux 增强功能
+
+管理 SELinux 策略（相当于使用通常在单个服务器上调用的 `semodule` ）不是由 SPO 本身完成的，
+而是由另一个名为 selinuxd 的容器完成，以提供更好的隔离。此版本切换到使用 selinuxd 
+容器从个人仓库到位于 [我们团队的 quay.io 存储库](https://quay.io/organization/security-profiles-operator)
+下的图像。selinuxd 仓库也已移至 [容器 GitHub 组织](https://github.com/containers/selinuxd)。
```
Comment (Sea-n) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:132]: 圖像 -> 鏡像

And remove spaces around links.

```suggestion
容器从个人仓库到位于[我们团队的 quay.io 存储库](https://quay.io/organization/security-profiles-operator)下的鏡像。
selinuxd 仓库也已移至[容器 GitHub 组织](https://github.com/containers/selinuxd)。
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+SELinux directories from the nodes, which means that the selinuxd container
+must be running the same distribution as the cluster nodes. SPO defaults
+to using CentOS-8 based containers, but we also build Fedora based ones.
+If you are using another distribution and would like us to add support for
+it, please file [an issue against selinuxd](https://github.com/containers/selinuxd/issues).
+-->
+请注意，selinuxd 动态链接到 libsemanage 并从节点挂载 SELinux 目录，
+这意味着 selinuxd 容器必须与集群节点运行相同的发行版。SPO 默认使用基于 CentOS-8 的容器，
+但我们也构建基于 Fedora 的容器。如果你使用其他发行版并希望我们添加对它的支持，
+请提交 [an issue against selinuxd](https://github.com/containers/selinuxd/issues)。
```
Comment (Sea-n) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:145]:
```suggestion
请[針對 selinuxd 提交 issue](https://github.com/containers/selinuxd/issues)。
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+### AppArmor 支持
+
+此版本引入了对 AppArmor 的初始支持，允许用户通过使用新的 [AppArmorProfile](https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/deploy/base/crds/apparmorprofile.yaml)
+将 AppArmor 配置文件加载和卸载到集群节点中 deploy/base/crds/apparmorprofile.yaml) CRD。
+
+<!--
+To enable AppArmor support use the [enableAppArmor feature gate](https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/examples/config.yaml#L10) switch of your SPO configuration.
+Then use our [apparmor example](https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/examples/apparmorprofile.yaml) to deploy your first profile across your cluster.
+-->
+要启用 AppArmor 支持，请使用 SPO 配置的 [enableAppArmor 功能门](https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/examples/config.yaml#L10)开关。
```
Comment (Sea-n) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:211]: The term "feature gate" is translated as "特性门控".

Please refer to this page:
https://kubernetes.io/zh-cn/docs/contribute/localization_zh/

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+SELinux messages of a node. Checkout the [official
+documentation](https://github.com/kubernetes-sigs/security-profiles-operator/blob/71b3915/installation-usage.md#using-the-log-enricher)
+to learn more about it.
+-->
+
+亮点之一是我们现在能够记录 seccomp 和 SELinux 使用操作符的配置文件
+[日志丰富器](https://github.com/kubernetes-sigs/security-profiles-operator/blob/71b3915/installation-usage.md#log-enricher-based-recording)。
+这使我们能够减少配置文件记录所需的依赖项，使其具有 [auditd](https://linux.die.net/man/8/auditd) 
+或 [syslog](https://en.wikipedia.org/wiki/Syslog) （如 fallback） 在节点上运行。 
+通过使用 `ProfileRecording` CRD 及其对应的[标签选择器](/docs/concepts/overview/working-with-objects/labels)，
```
Comment (Sea-n) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:75]: This page have Mandarin version.
```suggestion
通过使用 `ProfileRecording` CRD 及其对应的[标签选择器](/zh-cn/docs/concepts/overview/working-with-objects/labels)，
```

**Review by windsonsea — COMMENTED** [review]
Do we really need to localize these words? That's a product name:
- Operator
- Security Profiles Operator
- security-profiles-operator

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+
+<!--
+Under the hood, to know what the workload is doing SPO installs a special
+permissive policy called [selinuxrecording](https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/deploy/base/profiles/selinuxrecording.cil)
+on startup which allows everything and logs all AVCs to `audit.log`.
+These AVC messages are scraped by the log enricher component and when
+the recorded workload exits, the policy is created.
+-->
+在后台，为了知道工作负载在做什么，SPO 安装了一个名为 [selinuxrecording](https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/deploy/base/profiles/selinuxrecording.cil)
+的特殊许可策略，在启动时允许一切并将所有 AVC 记录到 `audit.log`。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:171]:
```suggestion
的、限制宽松的策略，允许执行所有操作并将所有 AVC 记录到 `audit.log` 中。
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+这些 AVC 消息由日志丰富器组件抓取，当记录的工作负载退出时，该策略被创建。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:172]:
```suggestion
这些 AVC 消息由日志增强组件抓取，当所记录的工作负载退出时，该策略被创建。
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+#### `SELinuxProfile` CRD graduation
+
+An `v1alpha2` version of the `SelinuxProfile` object has been introduced. This
+removes the raw Common Intermediate Language (CIL) from the object itself and
+instead adds a simple policy language to ease the writing and parsing
+experience.
+-->
+#### `SELinuxProfile` CRD 毕业
+
+引入了 `SelinuxProfile` 对象的 `v1alpha2` 版本。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:184]:
```suggestion
我们引入了 `SelinuxProfile` 对象的 `v1alpha2` 版本。 
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+
+An `v1alpha2` version of the `SelinuxProfile` object has been introduced. This
+removes the raw Common Intermediate Language (CIL) from the object itself and
+instead adds a simple policy language to ease the writing and parsing
+experience.
+-->
+#### `SELinuxProfile` CRD 毕业
+
+引入了 `SelinuxProfile` 对象的 `v1alpha2` 版本。 
+这从对象本身中删除了原始的通用中间语言 (CIL)，而是添加了一种简单的策略语言来简化编写和解析体验。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:185]:
```suggestion
这个版本从对象中删除了原始的通用中间语言 (CIL)，并添加了一种简单的策略语言来简化编写和解析体验。
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+
+<!--
+Alongside, a `RawSelinuxProfile` object was also introduced. This contains a
+wrapped and raw representation of the policy. This was intended for folks to be
+able to take their existing policies into use as soon as possible. However, on
+validations are done here.
+-->
+此外，还引入了 `RawSelinuxProfile` 对象。这包含策略的包装和原始表示。
+这是为了让人们能够尽快使用他们现有的政策。但是，验证是在这里完成的。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:194]:
```suggestion
引入此对象是为了让人们能够尽快将他们现有的策略付诸实现。但是，策略的验证是在这里完成的。
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 11]
```diff
@@ -0,0 +1,281 @@
...
+
+<!--
+### AppArmor support
+
+This version introduces the initial support for AppArmor, allowing users to load and 
+unload AppArmor profiles into cluster nodes by using the new [AppArmorProfile](https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/deploy/base/crds/apparmorprofile.yaml) CRD.
+-->
+### AppArmor 支持
+
+此版本引入了对 AppArmor 的初始支持，允许用户通过使用新的 [AppArmorProfile](https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/deploy/base/crds/apparmorprofile.yaml)
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:204]:
```suggestion
0.4.0 版本引入了对 AppArmor 的初始支持，允许用户通过使用新的
[AppArmorProfile](https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/deploy/base/crds/apparmorprofile.yaml)
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 12]
```diff
@@ -0,0 +1,281 @@
...
+将 AppArmor 配置文件加载和卸载到集群节点中 deploy/base/crds/apparmorprofile.yaml) CRD。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:205]:
```suggestion
在集群节点中 deploy/base/crds/apparmorprofile.yaml) CRD 加载或卸载 AppArmor 配置文件。
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 13]
```diff
@@ -0,0 +1,281 @@
...
+additional `spo-metrics-client` cluster role (and binding) to retrieve the
+metrics from within the cluster. If you're using
+[OpenShift](https://www.redhat.com/en/technologies/cloud-computing/openshift),
+then we provide an out of the box working
+[`ServiceMonitor`](https://github.com/kubernetes-sigs/security-profiles-operator/blob/71b3915/installation-usage.md#automatic-servicemonitor-deployment)
+to access the metrics.
+-->
+### 指标
+
+操作员现在公开指标，在我们的新 [指标文档](https://github.com/kubernetes-sigs/security-profiles-operator/blob/71b3915/installation-usage.md#using-metrics)中有详细描述。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:230]:
```suggestion
Operator 现在能够公开在我们的新[指标文档](https://github.com/kubernetes-sigs/security-profiles-operator/blob/71b3915/installation-usage.md#using-metrics)中详细描述的指标。
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 14]
```diff
@@ -0,0 +1,281 @@
...
+example, we now maintain an internal [gRPC](https://grpc.io) API to communicate
+within the operator across different features. We also improved the performance
+of the log enricher, which now caches results for faster retrieval of the log
+data. The operator can be put into a more [verbose log mode](https://github.com/kubernetes-sigs/security-profiles-operator/blob/71b3915/installation-usage.md#set-logging-verbosity)
+by setting `verbosity` from `0` to `1`.
+-->
+#### 可调试性和稳健性
+
+除了所有这些新功能外，我们还决定在内部重组安全配置文件操作程序的部分内容，使其更易于调试和更稳健。
+例如，我们现在维护了一个内部 [gRPC](https://grpc.io) API，以便在操作员内部跨不同功能进行通信。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:252]:
```suggestion
例如，我们现在维护了一个内部 [gRPC](https://grpc.io) API，以便在 Operator 内部跨不同功能组件进行通信。
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 15]
```diff
@@ -0,0 +1,281 @@
...
+我们还提高了日志丰富器的性能，现在它可以缓存结果，以便更快地检索日志数据。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:253]:
```suggestion
我们还提高了日志增强组件的性能，现在它可以缓存结果，以便更快地检索日志数据。
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 16]
```diff
@@ -0,0 +1,281 @@
...
+操作员可以通过将 `verbosity` 从 `0` 设置为 `1` 进入更详细的日志模式(https://github.com/kubernetes-sigs/security-profiles-operator/blob/71b3915/installation-usage.md#set-logging-verbosity)。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:254]:
```suggestion
Operator 可以通过将 `verbosity` 设置从 `0` 改为 `1`，启用更详细的日志模式(https://github.com/kubernetes-sigs/security-profiles-operator/blob/71b3915/installation-usage.md#set-logging-verbosity)。
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 17]
```diff
@@ -0,0 +1,281 @@
...
+
+<!--
+We also print the used `libseccomp` and `libbpf` versions on startup, as well as
+expose CPU and memory profiling endpoints for each container via the
+[`enableProfiling` option](https://github.com/kubernetes-sigs/security-profiles-operator/blob/71b3915/installation-usage.md#enable-cpu-and-memory-profiling).
+Dedicated liveness and startup probes inside of the operator daemon will now
+additionally improve the life cycle of the operator.
+-->
+我们还在启动时打印使用过的 `libseccomp` 和 `libbpf` 版本，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:263]:
```suggestion
我们还在启动时打印所使用的 `libseccomp` 和 `libbpf` 版本，
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 18]
```diff
@@ -0,0 +1,281 @@
...
+并通过 [`enableProfiling` 选项](https://github.com/kubernetes-sigs/security-profiles-operator/blob/71b3915/installation-usage.md#enable-cpu-and-memory-profiling)
+公开每个容器的 CPU 和内存分析端点。操作器守护程序内部的专用活动和启动探测现在将额外改善操作器的生命周期。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:265]:
```suggestion
公开每个容器的 CPU 和内存性能分析端点。
Operator 守护程序内部的专用的存活态探测和启动探测现在能进一步改善 Operator 的生命周期管理。
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 19]
```diff
@@ -0,0 +1,281 @@
...
+
+Thank you for reading this update. We're looking forward to future enhancements
+of the operator and would love to get your feedback about the latest release.
+Feel free to reach out to us via the Kubernetes slack
+[#security-profiles-operator](https://kubernetes.slack.com/messages/security-profiles-operator)
+for any feedback or question.
+-->
+## 总结
+
+感谢你阅读这次更新。我们期待着操作器未来的改进，并希望得到你对最新版本的反馈。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:279]:
```suggestion
感谢你阅读这次更新。我们期待着 Operater 的未来改进，并希望得到你对最新版本的反馈。
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 20]
```diff
@@ -0,0 +1,281 @@
...
+欢迎通过 Kubernetes slack [#security-profiles-operator](https://kubernetes.slack.com/messages/security-profiles-operator)
+与我们联系。提出任何反馈或问题。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:281]:
```suggestion
与我们联系，提出任何反馈或问题。
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 21]
```diff
@@ -0,0 +1,284 @@
...
+dedicated container, which ships a custom bpf module on every node to collect
+the syscalls for containers. It even supports older Kernel versions which do not
+expose the [BPF Type Format (BTF)](https://www.kernel.org/doc/html/latest/bpf/btf.html) per
+default as well as the `amd64` and `arm64` architectures. Checkout
+[our documentation](https://github.com/kubernetes-sigs/security-profiles-operator/blob/71b3915/installation-usage.md#ebpf-based-recording)
+to see it in action. By the way, we now add the seccomp profile architecture of
+the recorder host to the recorded profile as well.
+-->
+
+###与 seccomp 有关的改进
```
Comment (zhangxyjlu) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:96]:
```suggestion
### 与 seccomp 有关的改进
```

File: content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md [hunk 22]
```diff
@@ -0,0 +1,284 @@
...
+### AppArmor support
+
+This version introduces the initial support for AppArmor, allowing users to load and 
+unload AppArmor profiles into cluster nodes by using the new [AppArmorProfile](https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/deploy/base/crds/apparmorprofile.yaml) CRD.
+-->
+### AppArmor 支持
+
+0.4.0 版本引入了对 AppArmor 的初始支持，允许用户通过使用新的
+[AppArmorProfile](https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/deploy/base/crds/apparmorprofile.yaml)
+在集群节点中 deploy/base/crds/apparmorprofile.yaml) CRD 加载或卸载 AppArmor 配置文件。
```
Comment (zhangxyjlu) [inline content/zh-cn/blog/_posts/2021-12-17-security-profiles-operator-v0.4.0/index.md:207]:
```suggestion
在集群节点中 CRD 加载或卸载 AppArmor 配置文件。
```

**Pull request merged**